podaac/subsetter/subset.py

```diff
@@ -256,14 +256,21 @@ def calculate_chunks(dataset):
 
     Returns
     -------
-    dict
+    dict or auto
         The chunk dictionary, where the key is the dimension and the
-        value is 4000.
+        value is 4000. If dict is empty return auto if all dtype.hasobject
+        is false.
     """
-    chunk_dict = {dim: 4000 for dim in dataset.dims
-                  if dataset.dims[dim] > 4000
-                  and len(dataset.dims) > 1}
-    return chunk_dict
+    chunk = {dim: 4000 for dim in dataset.dims
+             if dataset.dims[dim] > 4000
+             and len(dataset.dims) > 1}
+
+    if not chunk:
+        dataset_dtypes_hasobject = [dataset.variables[var].dtype.hasobject for var in dataset.variables]
```

Comment:
What's the purpose of this line?

Reply:
> What's the purpose of this line?

check all variables dtype.hasobject which is a bool, and if any are true we can't use auto chunk as would throw this exception.

https://github.com/dask/dask/blob/main/dask/array/core.py#L3004